x-pack/plugins/security_solution/server/lib/entity_analytics/asset_criticality/helpers.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { AssetCriticalityRecord } from '../../../../common/api/entity_analytics';
+import { RISK_SCORING_NORMALIZATION_MAX } from '../risk_score/constants';
+import { CriticalityModifiers } from './constants';
+
+/**
+ * Retrieves the criticality modifier for a given criticality level.
+ *
+ * @param criticalityLevel The criticality level for which to get the modifier.
+ * @returns The associated criticality modifier for the given criticality level.
+ */
+export const getCriticalityModifier = (
+  criticalityLevel?: AssetCriticalityRecord['criticality_level']
+): number | undefined => {
+  if (criticalityLevel == null) {
+    return;
+  }
+
+  return CriticalityModifiers[criticalityLevel];
+};
+
+/**
+ * Applies asset criticality to a normalized risk score using bayesian inference.
+ * @param modifier - The criticality modifier to apply to the score.
+ * @param score - The normalized risk score to which the criticality modifier is applied
+ *
+ * @returns The risk score with the criticality modifier applied.
+ */
+export const applyCriticalityToScore = ({
+  modifier,
+  score,
+}: {
+  modifier: number | undefined;
+  score: number;
+}): number => {
+  if (modifier == null) {
+    return score;
+  }
+
+  return bayesianUpdate({ max: RISK_SCORING_NORMALIZATION_MAX, modifiers: [modifier], score });
+};
+
+/**
+ * Updates a score with the given modifiers using bayesian inference.
+ * @param modifiers - The modifiers to be applied to the score.
+ * @param score - The score to modifiers are applied
+ * @param max - The maximum value of the score.
+ *
+ * @returns The updated score with modifiers applied
+ */
+export const bayesianUpdate = ({
+  max,
+  modifiers,
+  score,
+}: {
+  max: number;
+  modifiers: number[];
+  score: number;
+}) => {
+  const modifier = modifiers.reduce((acc, curr) => acc * curr, 1);
```

Comment:
General question to improve my understanding of this code, as it's my first time reviewing this logic (now that Apoorva's left😭 )
The `modifiers` could be a few, e.g. `[0.75, 1.0, 1.0, 0.80]` and the end result of `modifier` would basically be those numbers all multiplied: `0.6`? What are some examples of these various multipliers?

Reply:
If multiple modifiers were provided, it would act as you describe. However, we currently only use this with a single modifier (from `getCriticalityModifier`).

I think I was imagining that multiple modifiers would come from different contexts, but those could be multiplied into a single modifier outside of this function, so I'm not sure there's a need for this code currently.

Reply:
Yeah, it seems safe to remove, and if needed in the future could be multiplied outside this function 🤔

Reply:
[0f236ff](https://github.com/elastic/kibana/pull/172417/commits/0f236ff1c5484b13bcc9dc65d68ac6002f0928f2)